src/test/scala/latis/input/TestHapiJsonAdapter.scala

```diff
@@ -0,0 +1,31 @@
+package latis.input
+
+import java.net.URI
+
+import latis.metadata.Metadata
+import latis.model._
+import latis.output.TextWriter
+import org.junit.Test
+import org.scalatest.junit.JUnitSuite
+
+class TestHapiJsonAdapter extends JUnitSuite {
+
+  @Test
+  def testHapiJson = {
+    val reader = new AdaptedDatasetReader {
+      //def uri: URI = new URI("https://cdaweb.gsfc.nasa.gov/hapi/data?id=AC_H0_MFI&time.min=2019-01-01&time.max=2019-01-02&parameters=Magnitude,dBrms&format=json")
+      def uri: URI = new URI(s"file:${System.getProperty("user.home")}/git/latis3-hapi/src/test/resources/data/hapi_json_data.txt")
+      def model: DataType = Function(
+        Scalar(Metadata("id" -> "Time", "type" -> "string")),
+        Tuple(
+          Scalar(Metadata("id" -> "Magnitude", "type" -> "float")),
+          Scalar(Metadata("id" -> "dBrms", "type" -> "float"))
+        )
+      )
+      def adapter = new HapiJsonAdapter(model)
+    }
+
+    val ds = reader.getDataset
+    TextWriter().write(ds) //TODO: assert against values in the dataset?
```

Comment:
How are we feeling about asserting against values here? This would be a true “unit test” if we continued hitting the local file instead of the live URL. But I’m not sure we want to keep that local data file.

Reply:
I don't see an issue with keeping test data. However, if we have data in `src/test/resources`, we should be able to use `getResource` and friends rather than using a path.

Reply:
We could probably get away with a smaller file, though. We only really need a few records.

Reply:
In general, I'd like to generate mock data in v3 instead of managing data files or hitting live data sources. We should separate out some integration tests for the latter (another day).

We could get creative with generating a generic dataset, encoding it with a HapiJsonEncoder, then test against that, but there are a number of hoops to jump through until we get there. So the question becomes, how do we generate the data. The easy thing would be to hardcode the content as a string in the test code, but then you might as well factor out that test data into a file which puts us back to where we started. So, I think the data file is fine (but I'd reduce it to a few samples). The fact that we are reading it in directly in the test code as opposed to using the latis machinery to construct a complete Dataset makes it sufficiently unit-like. We could refactor this to construct the HapiJsonAdapter then directly invoke it with the URI and test the `SampledFunction` we get back (instead of a complete Dataset). You could get the first sample from it. You can do an `unsafeForce` on it to get a `MemoizedFunction` from which you can directly access a Seq of Samples. (We still need some things like "head" and "take" to make this more useful.)

I think matching all the values in the first Sample would be a sufficient test.

Reply:
Unit test written with FlatSpec.